### PR TITLE
[stable/opa] Opa disable side effect webhook

### PR DESCRIPTION
**Pull request by nullck**
#### What this PR does / why we need it:

Some users that are using old version of Kubernetes (e.g 1.10.x) needs to deal with this when they try install the stable/opa chart:

```
Comparing opa stable/opa
********************

	Release was not present in Helm.  Diff will show entire contents as new.

********************
error validating "": error validating data: ValidationError(ValidatingWebhookConfiguration.webhooks[0]): unknown field "sideEffects" in io.k8s.api.admissionregistration.v1beta1.Webhook

in clusters/americas/helmfile.yaml: failed processing release opa: helm exited with status 1:
  Error: error validating "": error validating data: ValidationError(ValidatingWebhookConfiguration.webhooks[0]): unknown field "sideEffects" in io.k8s.api.admissionregistration.v1beta1.Webhook
  Error: plugin "diff" exited with error
  Error: plugin "tiller" exited with error
```

#### Which issue this PR fixes
Make the sideEffects optional to the end users, they just need to comment the line in the values.yaml file:

```
SideEffectClass for the webhook, setting to None enables dry-run
```
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/opa/templates/webhookconfiguration.yaml [hunk 1]
```diff
@@ -31,8 +31,9 @@ webhooks:
       service:
         name: {{ template "opa.fullname" . }}
         namespace: {{ .Release.Namespace }}
+{{ if .Values.admissionControllerSideEffect }}
```
Comment (Aisuko) [inline stable/opa/templates/webhookconfiguration.yaml]:
```
{{- if .Values.admissionControllerSideEffect }}
{{- end }}
```
Could make sense.

File: stable/opa/values.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ admissionControllerNamespaceSelector:
   matchExpressions:
     - {key: openpolicyagent.org/webhook, operator: NotIn, values: [ignore]}
 
-# SideEffectClass for the webhook, setting to None enables dry-run
+# SideEffectClass for the webhook, setting to None enables dry-run, if you want to disable the admissionControllerSideEffect option just comment the line below
 admissionControllerSideEffect: Unknown
```
Comment (Aisuko) [inline stable/opa/values.yaml:54]: What's the mean of this? The default value is `Unknown `? , Is there need some more explains?

**Review by Aisuko — CHANGES_REQUESTED** [review]
Please bump up the chart version. That can let your PR pass the CI. Thanks.

File: stable/opa/values.yaml (hunk 2 above)
Comment (tsandall) [reply stable/opa/values.yaml:54]: The field is not available before Kubernetes v1.12 so validation with older versions is generating errors. Including this detail (with the version) in the comment would be nice.

**Comment by Aisuko** [comment]
@nullck Please make signoff first.

**Comment by nullck** [comment]
> @nullck Please make signoff first.

sure, thanks

**Comment by nullck** [comment]
> > @nullck Please make signoff first.
> 
> sure, thanks

@Aisuko  done

**Comment by Aisuko** [comment]
@nullck Thanks, `git commit -s --amend && git push -f <your branch>` the command could trigger the DCO checker again.

**Comment by nullck** [comment]
> @nullck Thanks, `git commit -s --amend && git push -f <your branch>` the command could trigger the DCO checker again.

```
git commit -s --amend && git push -f origin opa-disable-SideEffect-Webhook                                                                                                                                                                                                                                                                               [23:50:39]
[opa-disable-SideEffect-Webhook 3fbd339c8] Merge branch 'opa-disable-SideEffect-Webhook' of github.com:nullck/charts into opa-disable-SideEffect-Webhook
 Date: Thu Sep 19 23:01:41 2019 -0300
Enter passphrase for key '/Users/nullck/.ssh/id_rsa':
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 286 bytes | 286.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0)
To github.com:nullck/charts.git
 + 53265ac9b...3fbd339c8 opa-disable-SideEffect-Webhook -> opa-disable-SideEffect-Webhook (forced update)
```

done

**Comment by Aisuko** [comment]
Please push to your PR, the PR's `DCO` status was still failing.

**Pull request closed without merging**